src/main/resources/public/ts/models/teacher/Eleve.ts

```diff
@@ -60,15 +60,20 @@ export class Eleve extends DefaultEleve  {
         return this.hasOwnProperty("displayName") ? this.displayName : this.firstName+" "+this.lastName;
     }
 
-    getMoyenne (devoirs?) : Promise<any> {
+    getMoyenne (devoirs, idEtablissement, idClasse, idPeriode, idMatiere?) : Promise<any> {
         return new Promise((resolve, reject) => {
             if (devoirs) {
-                let idDevoirsURL = "";
+                let idDevoirsURL:String = "";
```

Comment:
N'utilise pas l'objet `String`, utilise `string `

https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html